capplets/appearance/appearance-main.c

```diff
@@ -204,9 +206,15 @@ main (int argc, char **argv)
   if (wallpaper_files && !install_filename && !start_page)
     start_page = g_strdup ("background");
 
-  GtkNotebook* nb = GTK_NOTEBOOK(appearance_capplet_get_widget (data, "main_notebook"));
+  nb = GTK_NOTEBOOK (appearance_capplet_get_widget (data, "main_notebook"));
   gtk_widget_add_events (GTK_WIDGET (nb), GDK_SCROLL_MASK);
-  g_signal_connect (GTK_WIDGET (nb), "scroll-event",
+  g_signal_connect (nb, "scroll-event",
+                    G_CALLBACK (capplet_dialog_page_scroll_event_cb),
+                    GTK_WINDOW (w));
```

Comment:
```C
g_signal_connect(instance, detailed_signal, c_handler, data)
```
https://developer.gnome.org/gobject/stable/gobject-Signals.html#g-signal-connect
- instance = nb, it shoul be `GtkWidget*` bu it's `GtkNotebook*`, see 1st argument of `c_handler`
- c_handler = capplet_dialog_page_scroll_event_cb

```C
capplet_dialog_page_scroll_event_cb (GtkWidget *widget, GdkEventScroll *event, GtkWindow *window)
```